### DISCUSSION TITLE
Pause habit from day x to y

### DISCUSSION THREAD

**Discussion by Jomosoto** [iSoron/uhabits#815] [Feature Requests]
It would be useful if you can set, that from day x to y, the habit will not affect the percentage etc. For example habits like "learn for school" don't have to be done in the holidays and you could disable them for that time

**Comment by flashmaddison** [comment]
I signed in to suggest the same thing! I have a study habit for uni, but my term only runs October - July. I'd like to be able to pause the study habit during this time : )

**Comment by ke352802081770314** [comment]
indeed, a "vacation calendar" would solve some feature requests I've already read. The question would be, should this be maintained per habit, or would it perhaps be better to maintain one or more calendars where days can be switched on or off globally and then pick the one to apply to each habit in the habit editor?

**Reply by Jomosoto** [reply]
I like the second one

**Comment by WickyNilliams** [comment]
I would also like this feature. This would be extremely useful for vacations, illness etc, without having to "lie" to the app about doing the habit.

I think introducing a concept of vacation calendars not only complicates things, but also is quite a narrow view of the problem needing solving. A simple pause function would suffice for all use cases, right? You could choose some pre-defined times (pause for X days, or until a specific date), or until further notice. The UI could allow you to pick specific habits or all at once

**Reply by ke352802081770314** [reply]
I am puzzled why you consider it a narrow view. In my opinion introducing such a calendar is a much wider view than what you are describing as an alternative. The only thing that may make it seem narrow is the name I picked, "vacation" but I think to most people this is more intuitively understood than "pause calendar" or some other more accurate but technical name.

Your suggested pause function, as I understand it, will only allow me to pause "starting now" and therefore not scheduling in advance. Also, I would have to do it for each affected habit for every pause period again. Therefore, I am trading not having an additional and admittedly complex interface against much more work each time I want to use it. Plus I am losing the opportunity to correct records if I do not maintain the app in real time, because "always starting now" does not permit adjusting for past days.

**Reply by WickyNilliams** [reply]
I guess yeah it was the wording. But I still think a calendar creates a more complex mental model. You have multiple calendars which interact with the main habit calendar vs each habit being either in a paused or active state. Configuring individual calendars is more work than simply hitting a pause/play button. I don't think anything about the pausing approach precludes scheduling a pause.

That said, it could be that both features exist. Calendars or play/paused state can both be derived from a single data model. However I think the calendars are an advanced concept and the main UI should be a simpler play/pause
